test/unit/org/apache/cassandra/io/sstable/metadata/MetadataSerializerTest.java

```diff
@@ -85,7 +86,9 @@ public void testSerialization() throws IOException
 
             for (MetadataType type : MetadataType.values())
             {
-                assertEquals(originalMetadata.get(type), deserialized.get(type));
+                if ((type != MetadataType.STATS) || latestVersion.hasImprovedMinMax())
```

Comment:
is it needed?

Reply:
It's the same problem as before where `first` and `last` are null for `nb`. Let's solve that underlying issue first and then we'll know if we still need this.

Reply:
got it, it is another consequence of the fact that getLatestVersion does not return the latest version